### PR TITLE
SDKS-2381: Initial cache loading

### PR DESCRIPTION
**Pull request by javrudsky**
# iOS SDK

## What did you accomplish?
- Loading cache data on init
- Updated UT and Integration

### CODE REVIEW

File: Split/Network/Streaming/Synchronizer.swift [hunk 1]
```diff
@@ -178,4 +199,43 @@ class DefaultSynchronizer: Synchronizer {
...
+
+        let splitsInCache = splitsStorage.getAll()
+        var toDelete = [String]()
+
+        for split in splitsInCache {
+            guard let splitName = split.name else {
+                continue
+            }
+
+            if !namesToKeep.contains(splitName) {
```
Comment (EmilianoSanchez) [inline Split/Network/Streaming/Synchronizer.swift:222]: check this logic. 
I thinks it is wrongly deleting a split if it doesn't matches any name but matches a prefix.  

To delete a split, it shouldn't match neither with a name or a prefix. Something like:
```
if !namesToKeep.contains(splitName) && (!hasPrefix(for: splitName) || (let prefix = getPrefix(for: splitName) && !prefixesToKeep.contains(prefix)) ) {
    toDelete.append(splitName)
}
```

**Review by EmilianoSanchez — COMMENTED** [review]
Javi, check the comment regarding the filter condition for removing splits from cache

File: Split/Network/Streaming/Synchronizer.swift (hunk 1 above)
Comment (javrudsky) [reply Split/Network/Streaming/Synchronizer.swift]: Yup Doc, you're right! I've fixed the logic. Thank you!

**Review by EmilianoSanchez — APPROVED** [review]
LGTM

**Pull request merged**